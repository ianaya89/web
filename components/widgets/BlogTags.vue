<template>
  <div class="blog__tags py-4 mt-8">
    <p class="text-p2 uppercase font-medium text-grey border-b border-grey-l3 pb-4 mb-4">
      {{ $t('blog.headings.sections.tags') }}
    </p>

    <div class="flex flex-wrap">
      <blog-tag-link v-for="tag in tags" :key="tag.id" :tag="tag" />

      <p v-if="tags.length === 0" class="no-results">
        {{ $t('errors.no-tags') }}
      </p>
    </div>
  </div>
</template>

<script>
import BlogTagLink from '@/components/BlogTagLink'
import { TagsQuery } from '@/services/graphql/index'

export default {
  components: {
    BlogTagLink
  },

  apollo: {
    tags: TagsQuery
  },

  data: () => ({
    tags: TagsQuery.mock
  })
}
</script>
