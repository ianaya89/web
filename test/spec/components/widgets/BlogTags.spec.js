import { shallowMount } from '@vue/test-utils'
import BlogTags from '@/components/widgets/BlogTags'

describe('BlogTags', () => {
  it('renders BlogTags', () => {
    const wrapper = shallowMount(BlogTags)

    expect(wrapper.element).toMatchSnapshot()
  })
})
