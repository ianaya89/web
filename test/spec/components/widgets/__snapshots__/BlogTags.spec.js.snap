// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogTags renders BlogTags 1`] = `
<div
  class="blog__tags py-4 mt-8"
>
  <p
    class="text-p2 uppercase font-medium text-grey border-b border-grey-l3 pb-4 mb-4"
  >
    
    TRANSLATE:blog.headings.sections.tags
  
  </p>
   
  <div
    class="flex flex-wrap"
  >
    <blog-tag-link-stub
      tag="[object Object]"
    />
    <blog-tag-link-stub
      tag="[object Object]"
    />
    <blog-tag-link-stub
      tag="[object Object]"
    />
    <blog-tag-link-stub
      tag="[object Object]"
    />
    <blog-tag-link-stub
      tag="[object Object]"
    />
     
    <!---->
  </div>
</div>
`;
